pkg/upgrade/control_plane.go

```diff
@@ -617,14 +573,28 @@ func (u *ControlPlaneUpgrader) reconcileMachineAnnotations(log logr.Logger, mach
 	}
 
 	log.Info("Patching machine annotations")
-
 	if err := helper.Patch(context.TODO(), machine); err != nil {
 		return errors.Wrap(err, "error patching machine annotations")
 	}
 
 	return nil
 }
 
+func (u *ControlPlaneUpgrader) isReplacementMachine(machine *clusterv1.Machine) bool {
+	log := u.log.WithValues(
+		"machine", fmt.Sprintf("%s/%s", machine.Namespace, machine.Name),
+		"upgrade-id", u.upgradeID,
+	)
+	baseName := machine.Annotations[AnnotationMachineNameBase]
+	if baseName == "" {
```

Comment:
Would it be possible to put this validation before calling methods that need this annotation? That way we'd only have to check it once and we can error right away.

Reply:
TBH I don't think we need this check any more. I just refactored some previous logic into this method and left it here because I wanted to minimize unnecessary changes. For this case, if baseName was empty, the next condition check would be false which is the behavior we want anyways. 
Also this log would only serve as a distraction IMO.